Networking/Networking/Remote/InAppPurchasesRemote.swift

```diff
@@ -15,6 +15,25 @@ public class InAppPurchasesRemote: Remote {
         enqueue(request, mapper: mapper, completion: completion)
     }
 
+    /// Checks the WPCOM billing system for whether or not an In-app Purchase transaction has been handled
+    /// Unhandled transactions are those which have not yet marked as finished, in the Transactions.updates queue
+    /// We'll return the Site ID the transaction has been handled for, or an error if hasn't been handled yet
+    ///
+    /// - Parameters:
+    ///     - siteID: Site which the transaction has been handled for
+    ///     - transactionID: The transactionID of the specific transaction (not originalTransactionID)
+    ///     - completion: Closure to be executed upon completion. Returns the siteID of the transaction if success, or an error otherwise.
+    ///
+    public func checkTransactionHandled(with transactionID: UInt64,
```

Comment:
Not a blocker, I would refine this function name a bit, something like `checkIfTransactionIsHandled`

Reply:
Thanks! I thought about that, but it felt that with that name the signature should return a `Bool` rather than an `Int64` 🤔 I'll leave it as it is for the moment, but since we might modify the signature when implementing the rest of the feature this could change as well.

Reply:
Ah thanks for the answer. Then I would rename it to something that captures that meaning, like `retrieveHandledTransactionSiteID`. 

On a second look, I have a couple of questions about this method:

- There's documentation for the siteID parameter, but it's not part of the function
- We mention that unhandled transactions are are those which have not yet marked as finished, in the Transactions.updates queue. If so, why do we need to ask the backend for them? Can't we use the StoreKit API to know if they're handled or not?
- I don't think we should mention StoreKit implementation details `Transactions.updates` in this `Remote` class

Reply:
> Then I would rename it to something that captures that meaning, like retrieveHandledTransactionSiteID.

Thanks! Much better! I've renamed it in the following task, [here](https://github.com/woocommerce/woocommerce-ios/pull/10247/commits/47704895fc20937ddba7d2877e59a149dc462345)

> There's documentation for the siteID parameter, but it's not part of the function

Good catch, it seems to come from my initial implementation and then was not removed. Will amend [here](https://github.com/woocommerce/woocommerce-ios/pull/10247/commits/1d00daf5c49de03e9b11d8cc49437f517ecf40f6).

> We mention that unhandled transactions are are those which have not yet marked as finished, in the Transactions.updates queue. If so, why do we need to ask the backend for them? Can't we use the StoreKit API to know if they're handled or not?

I'm not 100% sure, but is my understanding that we can't because at the moment we're marking everything as `.finish()`, so for the StoreKit API all of them will be handled, since we send all transactions directly to `iap/orders`: If the site hasn't been upgraded yet then we create a new order and make the changes, if can't be updated then we return an error from the API, ignore it in the app as "we've already tackled this", and mark the transaction as finished. 

The problem is that in theory we should use this call only "_to complete a transaction after you deliver the purchased content or enable the purchased service"_ but at the moment we're doing it for all transactions, and ignoring some potential errors that could come from it, for example a failed transaction where has been paid on Apple end, but hasn't been completed in our end (we would still mark it as finished despite not upgrading the site) or an error caused when a merchant switches plans in Apple's end directly since M2.

> I don't think we should mention StoreKit implementation details Transactions.updates in this Remote class

You're right, amended in the following task with the other documentation [here](https://github.com/woocommerce/woocommerce-ios/pull/10247/commits/1d00daf5c49de03e9b11d8cc49437f517ecf40f6)